ResearchKit/Common/ORKStepViewController.m

```diff
@@ -300,10 +309,33 @@ - (void)goBackward {
     UIAccessibilityPostNotification(UIAccessibilityScreenChangedNotification, nil);
 }
 
+- (void)skip {
+    if (self.isBeingReviewed && !self.readOnlyMode) {
+        UIAlertController *alert = [UIAlertController alertControllerWithTitle:nil
+                                                                       message:nil
+                                                                preferredStyle:UIAlertControllerStyleActionSheet];
+        [alert addAction:[UIAlertAction actionWithTitle:ORKLocalizedString(@"BUTTON_CLEAR_ANSWER", nil)
+                                                  style:UIAlertActionStyleDestructive
+                                                handler:^(UIAlertAction *action) {
+                                                    dispatch_async(dispatch_get_main_queue(), ^{
+                                                        [self skipForward];
+                                                    });
+                                                }]];
+        [alert addAction:[UIAlertAction actionWithTitle:ORKLocalizedString(@"BUTTON_CANCEL", nil)
+                                                  style:UIAlertActionStyleCancel
+                                                handler:nil
+                          ]];
+        [self presentViewController:alert animated:YES completion:nil];
+    } else {
+        [self skipForward];
+    }
+}
+
 - (void)skipForward {
     [self goForward];
 }
```

Comment:
instead of changing the action selector to skip, lets keep it to skipForward. Instead, put the if statement inside skipForward, and call [self skip] or else [self goForward].